cirq/ops/pauli_string.py

```diff
@@ -111,38 +154,33 @@ def get(self, key: raw_types.Qid, default=None):
         return self._qubit_pauli_map.get(key, default)
     # pylint: enable=function-redefined
 
-    def __mul__(self, other):
-        if isinstance(other, numbers.Number):
-            return PauliString(self._qubit_pauli_map,
-                               self._coefficient * complex(other))
-        if isinstance(other, PauliString):
-            s1 = set(self.keys())
-            s2 = set(other.keys())
-            extra_phase = 1
-            terms = {}
-            for c in s1 - s2:
-                terms[c] = self[c]
-            for c in s2 - s1:
-                terms[c] = other[c]
-            for c in s1 & s2:
-                f, p = self[c].phased_pauli_product(other[c])
-                extra_phase *= f
-                if p != common_gates.I:
-                    terms[c] = p
-            return PauliString(
-                terms, self.coefficient * other.coefficient * extra_phase)
-        return NotImplemented
+    def __mul__(self, other) -> 'PauliString':
+        if not (isinstance(other, (PauliString, numbers.Number)) or
+                gate_operation.op_gate_isinstance(other,
+                                                  common_gates.IdentityGate)):
+            return NotImplemented
 
-    def __rmul__(self, other):
+        return PauliString(other,
```

Comment:
again, it seems very odd to rely on the (imo) strange constructor semantics to implement multiplication

Reply:
This implementation also kinda caught me by surprise. I tried a few different ways of writing it (e.g. having a module level method that both `__init__` and `__mul__` delegate to) but found this to be the clearest.